contentful/apps
# [INTEG-893] Properly handle languages for localized fields

## Purpose

There are a couple of issues currently in the modal for localized fields:
- If you try to apply output to a field that doesn't have the localized language or when where there is not content already, you get an error and cannot apply the output text
- We have decided that we want to remove the source/target language fields and instead display all of the fields and their language options within the field selectors

This is what the new field selector options look like:
![Screenshot 2023-07-31 at 8 59 55 AM](https://github.com/contentful/apps/assets/62958907/edae6d76-32c5-49f3-a07f-f98303f2ae21)

## Approach

Within the modal we have access to the entry from `cma.entry.get()` and the contentType from `cma.contentType.get()`. However, the fields array on the entry only contains fields that have content, and the fields array on the contentType only shows whether a field is localized. Therefore, we currently do not have a way of seeing all of the language options for a field within the modal. The Dialog location does not support `sdk.entry.fields` unfortunately. So in order for us to have access to all the language options for all the fields, I am getting this information from the Sidebar SDK and passing it via invocation parameters to the Dialog. I am definitely open to other approaches if anyone has any ideas!

I also wanted to have the order of the fields in the dropdown be in the same order as they appear on on the entry, which is the default locale first, and then the rest alphabetically by the full language name.

## Testing steps

Ensure you have localization turned on and have some fields that have several localization language options. You should now see all of the fields and their locale options in the field selectors. These should be ordered the same way that the fields are ordered on the entry. You should also be able to apply the output to fields consistently without errors (check when a field doesn't have content and when it's not the default locale). Test each of the actions and they should work as expected.
